parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java

```diff
@@ -608,6 +611,11 @@ private void addRowGroup(
         metaData.setSize_statistics(toParquetSizeStatistics(columnMetaData.getSizeStatistics()));
       }
 
+      if (columnMetaData.getGeospatialStatistics() != null) {
+        metaData.setGeospatial_statistics(
+            toParquetGeospatialStatistics(columnMetaData.getGeospatialStatistics()));
+      }
```

Comment:
It is better to test if the geospatial statistics is valid here, and skip writing it if the geospatial statistics is invalid.